ui/app/common/bifrost/resources/StaticLogResource.js

```diff
@@ -0,0 +1,66 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014-2020 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+(function () {
+  'use strict';
+
+  const module = angular.module('pnc.common.bifrost.resources');
+
+  module.value('STATIC_LOG_TEXT_PATH', '/text');
+
+  /**
+   * @author Dennis Zhou
+   */
+  module.factory('StaticLogResource', [
+    '$http',
+    'bifrostConfig',
+    'STATIC_LOG_TEXT_PATH',
+    function ($http, bifrostConfig, STATIC_LOG_TEXT_PATH) {
+      const ENDPOINT = bifrostConfig.getBifrostRestUrl() + STATIC_LOG_TEXT_PATH;
+
+      let resource = {};
+
+      resource.getStaticBuildLogText = (buildId) =>
+        $http({
```

Comment:
Why `$http` was preferred instead of `$resource`?

Reply:
The reason for using `$http` directly is because as Bifrost resource we are getting it by some matchFilters instead of resource id, this kind of usage can also be found on other resources that connect to no-id endpoints, such as GenericSetting(`ui/app/common/pnc-client/resources/GenericSetting.js`)